### PR TITLE
Make random number generator more standards conforming and hopefully threadsafe.

### PR DESCRIPTION
**Pull request by ursg**
This is lifted straight from the CUDA / OpenACC development branch, where the silly GPU compilers weren't able to cope with our regular random number generation mechanism.
Random generation is switched to std::defaut_random_engine, in the hope that it is implemented in a sane way on most platforms.
Also, by explicitly passing the random state object along with the random calls, the entire construction is threadsafe.

One reason why I am making this a separate pull request is that the changed random number behaviour causes testpackage diffs:
<details>
   <summary>Click to see testpackage output</summary>

```
running acctest_2_maxw_500k_100k_20kms_10deg 
Filtering is off and max number of Passes is =   0
--------------------------------------------------------------------------------------------
acctest_2_maxw_500k_100k_20kms_10deg  -  Verifying 3c10de608b6aae3c36be3be849568bf6f812f72c___DACC_SEMILAG_PQM__DTRANS_SEMILAG_PPM__DDP__DDPF__DVEC4D_AGNER__DACC_SEMILAG_PQM__DTRANS_SEMILAG_PPM__DDP__DDPF__DVEC4D_AGNER against current
--------------------------------------------------------------------------------------------
------------------------------------------------------------
 ref-time     |   new-time       |  speedup                |
------------------------------------------------------------
7.477        4.197         1.78151
------------------------------------------------------------
  variable     |     absolute diff     |     relative diff | 
------------------------------------------------------------
proton/vg_rho_0                1.49012e-08                 3.72525e-15    
proton/vg_v_0                4.08724e-07                 3.35585e-09    
proton/vg_v_1                3.73145e-07                 1.15968e-09    
proton/vg_v_2                3.25816e-07                 9.43798e-10    
--------------------------------------------------------------------------------------------
   Distribution function diff                                                               
--------------------------------------------------------------------------------------------
INFO Reading in two files.
File names: /mnt/btrfs/work/uganse/testpackage/testpackage_data/current/acctest_2_maxw_500k_100k_20kms_10deg/fullf.0000001.vlsv & /home/urs/vlasiator/testpackage/run_2021.10.13_14.18.40/acctest_2_maxw_500k_100k_20kms_10deg/fullf.0000001.vlsv
NonIdenticalBlocks:      0
IdenticalBlocks:         1573
Absolute_Error:          5.67245e-18
Mean-Absolute-Error:     1.94196e-22
Max-Absolute-Error:      3.37543e-20
Absolute-log-Error:      1.0315e-06
Mean-Absolute-log-Error: 3.53134e-11
--------------------------------------------------------------------------------------------
running acctest_3_substeps 
Filtering is off and max number of Passes is =   0
--------------------------------------------------------------------------------------------
acctest_3_substeps  -  Verifying 3c10de608b6aae3c36be3be849568bf6f812f72c___DACC_SEMILAG_PQM__DTRANS_SEMILAG_PPM__DDP__DDPF__DVEC4D_AGNER__DACC_SEMILAG_PQM__DTRANS_SEMILAG_PPM__DDP__DDPF__DVEC4D_AGNER against current
--------------------------------------------------------------------------------------------
------------------------------------------------------------
 ref-time     |   new-time       |  speedup                |
------------------------------------------------------------
102.8        56.87         1.80763
------------------------------------------------------------
  variable     |     absolute diff     |     relative diff | 
------------------------------------------------------------
proton/vg_rho_0                5.37783e-05                 1.34444e-11    
proton/vg_v_0                0.217167                 0.00153983    
proton/vg_v_1                0.809456                 0.00782949    
proton/vg_v_2                0.634636                 0.00388612    
--------------------------------------------------------------------------------------------
   Distribution function diff                                                               
--------------------------------------------------------------------------------------------
INFO Reading in two files.
File names: /mnt/btrfs/work/uganse/testpackage/testpackage_data/current/acctest_3_substeps/fullf.0000001.vlsv & /home/urs/vlasiator/testpackage/run_2021.10.13_14.18.40/acctest_3_substeps/fullf.0000001.vlsv
NonIdenticalBlocks:      3
IdenticalBlocks:         2194
Absolute_Error:          4.85507e-11
Mean-Absolute-Error:     1.12998e-15
Max-Absolute-Error:      2.83643e-13
Absolute-log-Error:      433.879
Mean-Absolute-log-Error: 0.0100982
--------------------------------------------------------------------------------------------
running transtest_2_maxw_500k_100k_20kms_20x20 
Filtering is off and max number of Passes is =   0
--------------------------------------------------------------------------------------------
transtest_2_maxw_500k_100k_20kms_20x20  -  Verifying 3c10de608b6aae3c36be3be849568bf6f812f72c___DACC_SEMILAG_PQM__DTRANS_SEMILAG_PPM__DDP__DDPF__DVEC4D_AGNER__DACC_SEMILAG_PQM__DTRANS_SEMILAG_PPM__DDP__DDPF__DVEC4D_AGNER against current
--------------------------------------------------------------------------------------------
------------------------------------------------------------
 ref-time     |   new-time       |  speedup                |
------------------------------------------------------------
106.4        68.08         1.56287
------------------------------------------------------------
  variable     |     absolute diff     |     relative diff | 
------------------------------------------------------------
proton/vg_rho_0                35.698                 3.56999e-05    
proton/vg_v_0                0.597269                 1.19454e-06    
proton/vg_v_1                0.395605                 7.91209e-07    
proton/vg_v_2                1.11978e-10                 1.52898    
--------------------------------------------------------------------------------------------
   Distribution function diff                                                               
--------------------------------------------------------------------------------------------
INFO Reading in two files.
File names: /mnt/btrfs/work/uganse/testpackage/testpackage_data/current/transtest_2_maxw_500k_100k_20kms_20x20/fullf.0000001.vlsv & /home/urs/vlasiator/testpackage/run_2021.10.13_14.18.40/transtest_2_maxw_500k_100k_20kms_20x20/fullf
.0000001.vlsv
NonIdenticalBlocks:      0
IdenticalBlocks:         423200
Absolute_Error:          2.76858e-08
Mean-Absolute-Error:     3.08993e-15
Max-Absolute-Error:      3.74725e-13
Absolute-log-Error:      2171.49
Mean-Absolute-log-Error: 0.000242354
--------------------------------------------------------------------------------------------
running acctest_4_helium 
Filtering is off and max number of Passes is =   0
--------------------------------------------------------------------------------------------
acctest_4_helium  -  Verifying 3c10de608b6aae3c36be3be849568bf6f812f72c___DACC_SEMILAG_PQM__DTRANS_SEMILAG_PPM__DDP__DDPF__DVEC4D_AGNER__DACC_SEMILAG_PQM__DTRANS_SEMILAG_PPM__DDP__DDPF__DVEC4D_AGNER against current
--------------------------------------------------------------------------------------------
------------------------------------------------------------
 ref-time     |   new-time       |  speedup                |
------------------------------------------------------------
35.66        18.54         1.92341
------------------------------------------------------------
  variable     |     absolute diff     |     relative diff | 
------------------------------------------------------------
helium/vg_rho_0                0.000275078                 1.37539e-10    
helium/vg_v_0                1.24749                 0.00606062    
helium/vg_v_1                0.462617                 9.28643e-07    
helium/vg_v_2                0.0343814                 0.000112581    
--------------------------------------------------------------------------------------------
running acctest_5_proton_antiproton 
Filtering is off and max number of Passes is =   0
--------------------------------------------------------------------------------------------
acctest_5_proton_antiproton  -  Verifying 3c10de608b6aae3c36be3be849568bf6f812f72c___DACC_SEMILAG_PQM__DTRANS_SEMILAG_PPM__DDP__DDPF__DVEC4D_AGNER__DACC_SEMILAG_PQM__DTRANS_SEMILAG_PPM__DDP__DDPF__DVEC4D_AGNER against current
--------------------------------------------------------------------------------------------
------------------------------------------------------------
 ref-time     |   new-time       |  speedup                |
------------------------------------------------------------
0.3105        0.1722         1.80314
------------------------------------------------------------
  variable     |     absolute diff     |     relative diff | 
------------------------------------------------------------
proton/vg_rho_0                2.79397e-09                 6.98485e-16    
proton/vg_v_0                2.01323e-09                 3.04955e-10    
proton/vg_v_1                2.30766e-08                 1.38724e-09    
proton/vg_v_2                2.45572e-09                 1.08443e-09    
--------------------------------------------------------------------------------------------
   Distribution function diff                                                               
--------------------------------------------------------------------------------------------
INFO Reading in two files.
File names: /mnt/btrfs/work/uganse/testpackage/testpackage_data/current/acctest_5_proton_antiproton/fullf.0000001.vlsv & /home/urs/vlasiator/testpackage/run_2021.10.13_14.18.40/acctest_5_proton_antiproton/fullf.0000001.vlsv
NonIdenticalBlocks:      0
IdenticalBlocks:         1416
Absolute_Error:          4.40703e-19
Mean-Absolute-Error:     1.63133e-23
Max-Absolute-Error:      3.76615e-21
Absolute-log-Error:      2.11621e-08
Mean-Absolute-log-Error: 7.83344e-13
--------------------------------------------------------------------------------------------
running restart_write 
Filtering is off and max number of Passes is =   0
Parameter data file (sw1.dat) has 1 values
Parameter data file (sw1.dat) has 1 values
--------------------------------------------------------------------------------------------
restart_write  -  Verifying 3c10de608b6aae3c36be3be849568bf6f812f72c___DACC_SEMILAG_PQM__DTRANS_SEMILAG_PPM__DDP__DDPF__DVEC4D_AGNER__DACC_SEMILAG_PQM__DTRANS_SEMILAG_PPM__DDP__DDPF__DVEC4D_AGNER against current
--------------------------------------------------------------------------------------------
------------------------------------------------------------
 ref-time     |   new-time       |  speedup                |
------------------------------------------------------------
0.05276        0.03991         1.32197
------------------------------------------------------------
  variable     |     absolute diff     |     relative diff | 
------------------------------------------------------------
proton/vg_rho_0                0.000125979                 1.26015e-10    
proton/vg_v_0                7.75635e-06                 1.53818e-11    
proton/vg_v_1                5.38238e-06                 2.06538e-11    
proton/vg_v_2                1.77722e-11                 1.36389    
fg_b_0                0                 0    
fg_b_1                0                 0    
fg_b_2                0                 0    
fg_e_0                2.89834e-15                 1.15934e-11    
fg_e_1                7.75638e-15                 1.55137e-11    
fg_e_2                1.06547e-14                 4.26242e-11    
--------------------------------------------------------------------------------------------
running restart_read 
Filtering is off and max number of Passes is =   0
Parameter data file (sw1.dat) has 1 values
Parameter data file (sw1.dat) has 1 values
--------------------------------------------------------------------------------------------
restart_read  -  Verifying 3c10de608b6aae3c36be3be849568bf6f812f72c___DACC_SEMILAG_PQM__DTRANS_SEMILAG_PPM__DDP__DDPF__DVEC4D_AGNER__DACC_SEMILAG_PQM__DTRANS_SEMILAG_PPM__DDP__DDPF__DVEC4D_AGNER against current
--------------------------------------------------------------------------------------------
------------------------------------------------------------
 ref-time     |   new-time       |  speedup                |
------------------------------------------------------------
                 NA
------------------------------------------------------------
  variable     |     absolute diff     |     relative diff | 
------------------------------------------------------------
proton/vg_rho_0                0                 0    
proton/vg_v_0                0                 0    
proton/vg_v_1                0                 0    
proton/vg_v_2                0                 0    
fg_b_0                0                 0    
fg_b_1                0                 0    
fg_b_2                0                 0    
fg_e_0                0                 0    
fg_e_1                0                 0    
fg_e_2                0                 0    
--------------------------------------------------------------------------------------------
running Magnetosphere_small 
Filtering is off and max number of Passes is =   0
Parameter data file (sw1.dat) has 1 values
--------------------------------------------------------------------------------------------
Magnetosphere_small  -  Verifying 3c10de608b6aae3c36be3be849568bf6f812f72c___DACC_SEMILAG_PQM__DTRANS_SEMILAG_PPM__DDP__DDPF__DVEC4D_AGNER__DACC_SEMILAG_PQM__DTRANS_SEMILAG_PPM__DDP__DDPF__DVEC4D_AGNER against current
--------------------------------------------------------------------------------------------
------------------------------------------------------------
 ref-time     |   new-time       |  speedup                |
------------------------------------------------------------
483.2        268.4         1.8003
------------------------------------------------------------
  variable     |     absolute diff     |     relative diff | 
------------------------------------------------------------
proton/vg_rho_0                1109.03                 0.00110041    
proton/vg_v_0                2630.77                 0.00521494    
proton/vg_v_1                1858.61                 0.089269    
proton/vg_v_2                220.326                 0.0707499    
fg_b_0                1.32616e-11                 0.00332383    
fg_b_1                1.28398e-11                 0.00318687    
fg_b_2                2.8995e-10                 1.7397e-05    
fg_e_0                0.000280825                 0.121659    
fg_e_1                0.000641795                 0.100141    
fg_e_2                4.45102e-05                 0.0249384    
--------------------------------------------------------------------------------------------
   Distribution function diff                                                               
--------------------------------------------------------------------------------------------
INFO Reading in two files.
File names: /mnt/btrfs/work/uganse/testpackage/testpackage_data/current/Magnetosphere_small/bulk.0000001.vlsv & /home/urs/vlasiator/testpackage/run_2021.10.13_14.18.40/Magnetosphere_small/bulk.0000001.vlsv
NonIdenticalBlocks:      2
IdenticalBlocks:         3500
Absolute_Error:          4.63669e-10
Mean-Absolute-Error:     1.52302e-14
Max-Absolute-Error:      3.564e-12
Absolute-log-Error:      459.105
Mean-Absolute-log-Error: 0.0150803
--------------------------------------------------------------------------------------------
running Magnetosphere_polar_small 
Filtering is off and max number of Passes is =   0
Parameter data file (sw1.dat) has 1 values
--------------------------------------------------------------------------------------------
Magnetosphere_polar_small  -  Verifying 3c10de608b6aae3c36be3be849568bf6f812f72c___DACC_SEMILAG_PQM__DTRANS_SEMILAG_PPM__DDP__DDPF__DVEC4D_AGNER__DACC_SEMILAG_PQM__DTRANS_SEMILAG_PPM__DDP__DDPF__DVEC4D_AGNER against current
--------------------------------------------------------------------------------------------
------------------------------------------------------------
 ref-time     |   new-time       |  speedup                |
------------------------------------------------------------
554.6        307         1.80651
------------------------------------------------------------
  variable     |     absolute diff     |     relative diff | 
------------------------------------------------------------
proton/vg_rho_0                2918.67                 0.00188032    
proton/vg_v_0                6182.18                 0.00822246    
proton/vg_v_1                137.902                 0.18757    
proton/vg_v_2                3967.63                 0.0749571    
fg_b_0                5.6587e-10                 0.000279754    
fg_b_1                1.98197e-11                 0.277284    
fg_b_2                6.79272e-10                 1.29046e-06    
fg_e_0                3.00819e-05                 0.301245    
fg_e_1                0.000862048                 0.0722595    
fg_e_2                2.48476e-05                 0.234658    
proton/vg_v_nonthermal_0                43782.9                 0.0547543    
proton/vg_v_nonthermal_1                10662.6                 0.269036    
proton/vg_v_nonthermal_2                25719.6                 0.0532728    
proton/vg_ptensor_nonthermal_diagonal_0                8.40414e-14                 0.0310106    
proton/vg_ptensor_nonthermal_diagonal_1                1.00167e-13                 0.0199351    
proton/vg_ptensor_nonthermal_diagonal_2                9.24835e-14                 0.0188084    
--------------------------------------------------------------------------------------------
   Distribution function diff                                                               
--------------------------------------------------------------------------------------------
INFO Reading in two files.
File names: /mnt/btrfs/work/uganse/testpackage/testpackage_data/current/Magnetosphere_polar_small/bulk.0000001.vlsv & /home/urs/vlasiator/testpackage/run_2021.10.13_14.18.40/Magnetosphere_polar_small/bulk.0000001.vlsv
NonIdenticalBlocks:      192
IdenticalBlocks:         104233
Absolute_Error:          3.92678e-09
Mean-Absolute-Error:     2.69447e-15
Max-Absolute-Error:      4.52896e-12
Absolute-log-Error:      10596.2
Mean-Absolute-log-Error: 0.00727086
--------------------------------------------------------------------------------------------
running test_fp_substeps 
Filtering is off and max number of Passes is =   0
--------------------------------------------------------------------------------------------
test_fp_substeps  -  Verifying 3c10de608b6aae3c36be3be849568bf6f812f72c___DACC_SEMILAG_PQM__DTRANS_SEMILAG_PPM__DDP__DDPF__DVEC4D_AGNER__DACC_SEMILAG_PQM__DTRANS_SEMILAG_PPM__DDP__DDPF__DVEC4D_AGNER against current
--------------------------------------------------------------------------------------------
------------------------------------------------------------
 ref-time     |   new-time       |  speedup                |
------------------------------------------------------------
25.71        22.53         1.14115
------------------------------------------------------------
  variable     |     absolute diff     |     relative diff | 
------------------------------------------------------------
fg_b_0                3.33457e-24                 4.2821e-15    
fg_b_1                3.10193e-24                 3.98335e-15    
fg_b_2                3.07608e-24                 3.95016e-15    
fg_e_0                8.57621e-21                 4.94046e-15    
fg_e_1                8.47033e-21                 4.87947e-15    
fg_e_2                6.98802e-21                 4.02556e-15    
--------------------------------------------------------------------------------------------
running Flowthrough_trans_periodic 
Filtering is off and max number of Passes is =   0
--------------------------------------------------------------------------------------------
Flowthrough_trans_periodic  -  Verifying 3c10de608b6aae3c36be3be849568bf6f812f72c___DACC_SEMILAG_PQM__DTRANS_SEMILAG_PPM__DDP__DDPF__DVEC4D_AGNER__DACC_SEMILAG_PQM__DTRANS_SEMILAG_PPM__DDP__DDPF__DVEC4D_AGNER against current
--------------------------------------------------------------------------------------------
------------------------------------------------------------
 ref-time     |   new-time       |  speedup                |
------------------------------------------------------------
1.335        0.9891         1.34971
------------------------------------------------------------
  variable     |     absolute diff     |     relative diff | 
------------------------------------------------------------
proton/vg_rho_0                1.16415e-09                 4.69277e-16    
proton/vg_v_0                5.82077e-10                 1.17277e-15    
proton/vg_v_1                5.82077e-10                 1.17295e-15    
proton/vg_v_2                8.73115e-10                 2.18279e-15    
fg_b_0                0                 0    
fg_b_1                0                 0    
fg_b_2                0                 0    
fg_e_0                0                 0    
fg_e_1                0                 0    
fg_e_2                0                 0    
--------------------------------------------------------------------------------------------
running Flowthrough_x_inflow_y_outflow 
Filtering is off and max number of Passes is =   0
Parameter data file (sw1.dat) has 1 values
Parameter data file (sw1.dat) has 1 values
--------------------------------------------------------------------------------------------
Flowthrough_x_inflow_y_outflow  -  Verifying 3c10de608b6aae3c36be3be849568bf6f812f72c___DACC_SEMILAG_PQM__DTRANS_SEMILAG_PPM__DDP__DDPF__DVEC4D_AGNER__DACC_SEMILAG_PQM__DTRANS_SEMILAG_PPM__DDP__DDPF__DVEC4D_AGNER against current
--------------------------------------------------------------------------------------------
------------------------------------------------------------
 ref-time     |   new-time       |  speedup                |
------------------------------------------------------------
2.08        1.524         1.36483
------------------------------------------------------------
  variable     |     absolute diff     |     relative diff | 
------------------------------------------------------------
proton/vg_rho_0                0.00012598                 1.26016e-10    
proton/vg_v_0                7.75704e-06                 1.55151e-11    
proton/vg_v_1                2.96789e-06                 1.18716e-11    
proton/vg_v_2                5.30254e-12                 3.08861    
fg_b_0                0                 0    
fg_b_1                0                 0    
fg_b_2                0                 0    
fg_e_0                2.89834e-15                 1.15934e-11    
fg_e_1                7.75638e-15                 1.55137e-11    
fg_e_2                1.06547e-14                 4.26242e-11    
--------------------------------------------------------------------------------------------
running Flowthrough_x_inflow_y_outflow_acc 
Filtering is off and max number of Passes is =   0
Parameter data file (sw1.dat) has 1 values
Parameter data file (sw1.dat) has 1 values
--------------------------------------------------------------------------------------------
Flowthrough_x_inflow_y_outflow_acc  -  Verifying 3c10de608b6aae3c36be3be849568bf6f812f72c___DACC_SEMILAG_PQM__DTRANS_SEMILAG_PPM__DDP__DDPF__DVEC4D_AGNER__DACC_SEMILAG_PQM__DTRANS_SEMILAG_PPM__DDP__DDPF__DVEC4D_AGNER against current
--------------------------------------------------------------------------------------------
------------------------------------------------------------
 ref-time     |   new-time       |  speedup                |
------------------------------------------------------------
506.8        235.4         2.15293
------------------------------------------------------------
  variable     |     absolute diff     |     relative diff | 
------------------------------------------------------------
proton/vg_rho_0                14.5689                 6.96392e-06    
proton/vg_v_0                0.696173                 1.39243e-06    
proton/vg_v_1                0.170669                 6.82042e-07    
proton/vg_v_2                0.643969                 0.000569991    
fg_b_0                0                 -1    
fg_b_1                0                 0    
fg_b_2                0                 -1    
fg_e_0                6.213e-21                 0.341953    
fg_e_1                0                 -1    
fg_e_2                7.75699e-14                 1.55149e-11    
--------------------------------------------------------------------------------------------
running Flowthrough_amr 
Filtering is off and max number of Passes is =   0
Maximum refinement level is 2
Finished first level of refinement
Finished second level of refinement
Parameter data file (sw1.dat) has 1 values
--------------------------------------------------------------------------------------------
Flowthrough_amr  -  Verifying 3c10de608b6aae3c36be3be849568bf6f812f72c___DACC_SEMILAG_PQM__DTRANS_SEMILAG_PPM__DDP__DDPF__DVEC4D_AGNER__DACC_SEMILAG_PQM__DTRANS_SEMILAG_PPM__DDP__DDPF__DVEC4D_AGNER against current
--------------------------------------------------------------------------------------------
------------------------------------------------------------
 ref-time     |   new-time       |  speedup                |
------------------------------------------------------------
590.5        353.2         1.67186
------------------------------------------------------------
  variable     |     absolute diff     |     relative diff | 
------------------------------------------------------------
proton/vg_rho_0                2.84817e-05                 1.32111e-11    
proton/vg_v_0                1.74946e-05                 8.09117e-11    
proton/vg_v_1                1.31707e-05                 1.58453e-09    
proton/vg_v_2                8.70455e-06                 1.00716e-09    
fg_b_0                3.71135e-21                 3.71135e-12    
fg_b_1                7.15077e-21                 3.58621e-12    
fg_b_2                7.49819e-21                 3.62536e-12    
fg_e_0                3.69361e-16                 3.9552e-11    
fg_e_1                6.85433e-16                 1.67792e-12    
fg_e_2                8.03611e-16                 2.06581e-12    
--------------------------------------------------------------------------------------------

```
</details>

So this needs to be checked carefully, so that we do not again introduce a spurious entropy wave (through correlation in random number generation or something). Then hopefully, this will make checking GPU implementation diffs easier.

### CODE REVIEW

File: projects/project.cpp [hunk 1]
```diff
@@ -508,7 +494,7 @@ namespace projects {
       const CellID cellID = (int) ((x - Parameters::xmin) / dx) +
```
Comment (ykempf) [inline projects/project.cpp:494]: Who coded that? (Me, likely??) Why don't we just read the ID from CellParams::??

File: projects/Dispersion/Dispersion.cpp [hunk 2]
```diff
@@ -216,18 +216,15 @@ namespace projects {
       creal dy = cellParams[CellParams::DY];
       creal z = cellParams[CellParams::ZCRD];
       creal dz = cellParams[CellParams::DZ];
-      
-      CellID cellID = (int) ((x - Parameters::xmin) / dx) +
-         (int) ((y - Parameters::ymin) / dy) * Parameters::xcells_ini +
-         (int) ((z - Parameters::zmin) / dz) * Parameters::xcells_ini * Parameters::ycells_ini;
 
-      setRandomSeed(cellID);
+      std::default_random_engine rndState;
```
Comment (ykempf) [inline projects/Dispersion/Dispersion.cpp:220]: What is this thing? Is this going to guarantee the same random result for a given cell no matter what the task+thread layout is? (Comment valid for all further occurrences.)

File: projects/project.cpp (hunk 1 above)
Comment (ursg) [reply projects/project.cpp:494]: Yep, you committed that in 2014, I believe: https://github.com/ursg/vlasiator/commit/88cd0b4b7e413953d495d7d6003c9a4529720447#diff-ac171b423794277eb8b6b862f7037709aab19b0bdc5a38b974627241a30cb159R212

File: projects/Dispersion/Dispersion.cpp (hunk 2 above)
Comment (ykempf) [reply projects/Dispersion/Dispersion.cpp:220]: OK. Good I guess then?

**Comment by ursg** [comment]
Rebased this onto current dev (in part, to see what the CI testpackage says about it now).
Still produces significant diffs, but this is probably due to the nature of the changed random generator. So still undecided whether this can go in.

File: projects/MultiPeak/MultiPeak.cpp [hunk 3]
```diff
@@ -136,83 +136,104 @@ namespace projects {
...
-               for (uint vk=0; vk<N; ++vk) {
-                  creal VX = vx + 0.5*DVX + vi*DVX;
-                  creal VY = vy + 0.5*DVY + vj*DVY;
-                  creal VZ = vz + 0.5*DVZ + vk*DVZ;
-                  avg += getDistribValue(VX,VY,VZ,DVX,DVY,DVZ,popID);
-               }
+      // Real avgTotal = 0.0;
+      // bool ok = false;
+      // uint N = nVelocitySamples; // Start by using nVelocitySamples
+      // int N3_sum = 0;           // Sum of sampling points used so far
```
Comment (ykempf) [inline projects/MultiPeak/MultiPeak.cpp:156]: Drop commented chunks?

File: projects/project.cpp (hunk 1 above)
Comment (ykempf) [reply projects/project.cpp:493]: So, are we changing this or not bothering?

File: testpackage/tests/Flowthrough_trans_periodic/Flowthrough_trans_periodic.cfg [hunk 4]
```diff
@@ -48,15 +48,15 @@ mass_units = PROTON
...
+vx_min = -640000.0
+vx_max = +640000.0
+vy_min = -640000.0
+vy_max = +640000.0
+vz_min = -640000.0
+vz_max = +640000.0
+vx_length = 32
+vy_length = 32
+vz_length = 32
 [proton_sparse]
```
Comment (ykempf) [inline testpackage/tests/Flowthrough_trans_periodic/Flowthrough_trans_periodic.cfg:60]: This too.

File: testpackage/tests/Flowthrough_x_inflow_y_outflow/Flowthrough_x_inflow_y_outflow.cfg [hunk 5]
```diff
@@ -47,15 +47,15 @@ mass_units = PROTON
...
-vz_length = 15
+vx_min = -640000.0
+vx_max = +640000.0
+vy_min = -640000.0
+vy_max = +640000.0
+vz_min = -640000.0
+vz_max = +640000.0
+vx_length = 32
+vy_length = 32
+vz_length = 32
```
Comment (ykempf) [inline testpackage/tests/Flowthrough_x_inflow_y_outflow/Flowthrough_x_inflow_y_outflow.cfg:58]: ditto

**Pull request merged**